crates/biome_js_formatter/src/js/expressions/call_arguments.rs

```diff
@@ -873,11 +874,48 @@ fn should_group_last_argument(
 ///     HTMLElement     => true
 ///     string | object => false
 ///     Foo<string>     => false
-///
-/// Note that Prettier allows extracting the inner type from arrays and
-/// single-argument generic types, but for now, this implementation does not.
-fn is_simple_ts_type(ty: AnyTsType) -> bool {
-    match ty {
+/// This function also introspects into array and generic types to extract the
+/// core type, but only to a limited extent:
+///     string[]        => string
+///     string[][]      => string
+///     string[][][]    => string
+///     Foo<string>[][] => string
+///     Foo<string[]>[] => string[]
+///     Foo<string[][]> => string[][]
+fn is_simple_ts_type(ty: &AnyTsType) -> bool {
+    // Reach up to two-levels deep into array types:
+    //     string[]     => string
+    //     string[][]   => string
+    //     string[][][] => string[]
+    let extracted_array_type = match ty {
+        AnyTsType::TsArrayType(array) => match array.element_type() {
+            Ok(AnyTsType::TsArrayType(inner_array)) => inner_array.element_type().ok(),
+            Ok(element_type) => Some(element_type),
+            _ => None,
+        },
+        _ => None,
+    };
+
+    // Then, extract the first type parameter of a Generic as long as it's the
+    // only parameter:
+    //     Foo<number> => number
+    //     Foo<number, string> => Foo<number, string>
+    let extracted_generic_type = match extracted_array_type {
+        Some(AnyTsType::TsReferenceType(generic)) => {
+            generic.type_arguments().map_or(None, |type_arguments| {
+                let argument_list = type_arguments.ts_type_argument_list();
+                if argument_list.len() == 1 {
+                    argument_list.first().map_or(None, |first| first.ok())
```

Comment:
This looks a little weird, but it's unwrapping `Option<SyntaxResult<Node>>` into just `Option<Node>` to match the structure of the rest of the types here.